.github/workflows/create-deployment-spack.yml

```diff
@@ -0,0 +1,46 @@
+name: Create Deployment Spack
+on:
+  workflow_dispatch:
+    inputs:
+      spack-version:
+        type: string
+        required: true
+        description: A version of spack
+      spack-packages-version:
+        type: string
+        required: true
+        default: main
+        description: A version of ACCESS-NRI/spack-packages
+      spack-config-version:
+        type: string
+        required: true
+        default: main
+        description: A version of ACCESS-NRI/spack-config
+jobs:
+  create-spack:
+    name: Spack
+    runs-on: ubuntu-latest
+    environment: "Gadi Spack"
+    steps:
+      - name: Setup SSH
+        id: ssh
+        uses: access-nri/actions/.github/actions/setup-ssh@main
+        with:
+          hosts: ${{ secrets.SSH_HOST }}
+          private-key: ${{ secrets.SSH_KEY }}
+      - name: Install
+        env:
+          SPACK_VERSION_LOCATION: ${{ vars.ROOT_SPACK_LOCATION }}/${{ inputs.spack-version }}
+          SPACK_LOCATION: ${{ env.SPACK_VERSION_LOCATION }}/spack
+          SPACK_PACKAGES_LOCATION: ${{ env.SPACK_VERSION_LOCATION }}/spack-packages
+          SPACK_CONFIG_LOCATION: ${{ env.SPACK_VERSION_LOCATION }}/spack-config
+          SPACK_RELEASE_LOCATION: ${{ env.SPACK_VERSION_LOCATION }}/release
+        run: |
+          ssh ${{ secrets.SSH_USER }}@${{ secrets.SSH_HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
+          mkdir ${{ env.SPACK_VERSION_LOCATION }}
+          git clone -c feature.manyFiles=true https://github.com/spack/spack.git ${{ env.SPACK_VERSION_LOCATION }} --branch ${{ inputs.spack-version }} --single-branch --depth=1
+          git clone  https://github.com/ACCESS-NRI/spack-packages.git ${{ env.SPACK_VERSION_LOCATION }} --branch ${{ inputs.spack-packages-version }}
+          git clone  https://github.com/ACCESS-NRI/spack-config.git ${{ env.SPACK_VERSION_LOCATION }} --branch ${{ inputs.spack-config-version }}
```

Comment:
You've specified `--single-branch --depth=1` for `spack` but not the others. I assume this is because of the sheer size of the `spack` repo?

I don't have a problem with them being different, but is there a reason not to use those options with the other repos too?

Speaking of which, how do we update `spack-packages` and `spack-config` if required?

Reply:
Correct. 

I don't use them because we may want to swap to a different branch (`--single-branch` makes this a bit harder) for spack-packages/config for some reason, whereas with spack we would just create a new `.../spack/<version>` folder. 

For `spack-config`, we have an on-push workflow: https://github.com/ACCESS-NRI/spack-config/blob/main/.github/workflows/update-deployment-repos.yml
For `spack-packages` we don't have that workflow yet...I might have to create it!